---
title: "Bidirectional two-sample MR analysis"
output: rmarkdown::github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## GWAS summary statistics data

* HEM ([Zheng *et al.*](https://pubmed.ncbi.nlm.nih.gov/33888516/)) - GCST90014033
* DIV ([Schafmayer *et al.*](https://pubmed.ncbi.nlm.nih.gov/30661054/)) - GCST008105
* IBS ([Eijsbouts *et al.*](https://pubmed.ncbi.nlm.nih.gov/34741163/)) - GCST90016564

&nbsp;
```{r, message=FALSE, warning=FALSE}

# libraries
library(tidyverse)
library(TwoSampleMR)

```

### Download files
```{r, message=FALSE, warning=FALSE, eval=FALSE}

##
## download files
##

ebi_ftp <- "http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/"

# DIV - GCST008105
# https://pubmed.ncbi.nlm.nih.gov/30661054/
download.file( 
  url = paste0(ebi_ftp,"GCST008001-GCST009000/GCST008105/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz"),
  destfile = "data/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz",
  method = "curl"
)

# IBS - GCST90016564 
# https://pubmed.ncbi.nlm.nih.gov/34741163/
download.file( 
  url = paste0(ebi_ftp,"GCST90016001-GCST90017000/GCST90016564/GCST90016564_buildGRCh37.tsv"),
  destfile = "data/GCST90016564_buildGRCh37.tsv",
  method = "curl"
)
# compress
R.utils::gzip(
  "data/GCST90016564_buildGRCh37.tsv",
  overwrite=TRUE
)

# HEM - GCST90014033
# https://pubmed.ncbi.nlm.nih.gov/33888516/
# can be accessed directly from the TwoSampleMR

```

## MR analysis

&nbsp;

### Instrument variable selection 

For the instrument variable (IVs) selection thresholds, we used the `TwoSampleMR` workflow with the following parameters:

1) p-value = 5e^-08^ (if N~GWS loci~ > 10) or p-value = 1e^-05^ (if N~GWS loci~ < 10);
2) LD - r^2^ = 0.001 within 1Mb window (EUR);
3) MR-PRESSO horizontal pleiotropy removal (outlier p-value < 0.1)

&nbsp;

### SNPs -> HEM (exposure) -> IBS (outcome)
```{r, message=TRUE, warning=FALSE}

## get instruments

# from OpenGWAS db
HEM_exp_dat <- extract_instruments(
  "ebi-a-GCST90014033",
  r2 = 0.001,
  kb = 1000
  ) %>% mutate(
    exposure = "HEM (Zheng et al.)"
  )

IBS_out_dat <- read_outcome_data(
  snps = HEM_exp_dat$SNP,
  filename = "data/GCST90016564_buildGRCh37.tsv.gz",
  sep = "\t",
  snp_col = "variant_id",
  chr_col = "chromosome", 
  pos_col = "base_pair_location",
  beta_col = "beta",
  se_col = "standard_error",
  effect_allele_col = "effect_allele",
  other_allele_col = "other_allele",
  eaf_col = "effect_allele_frequency",
  pval_col = "p_value"
  ) %>% mutate( 
    outcome = "IBS (Eijsbouts et al.)"
  )

# check if any SNPs are missing
missing <- IBS_out_dat %>% 
  filter(!SNP %in% HEM_exp_dat$SNP) %>% 
  pull(SNP)

missing

# harmonize the exposure and outcome data
HEM_IBS_dat <- harmonise_data(
  HEM_exp_dat,
  IBS_out_dat
  )

# test for horizontal pleiotropy
mr_pleiotropy_test(HEM_IBS_dat) %>%
  select(-1:-2)

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
HEM_IBS_presso <- run_mr_presso(
  HEM_IBS_dat, 
  NbDistribution = 1000,
  SignifThreshold = 0.1
  )

# global test results
HEM_IBS_presso[[1]]$`MR-PRESSO results`$`Global Test`

# HP outliers
HEM_IBS_dat %>% 
  filter(
    row_number() %in% HEM_IBS_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    ) %>% 
  select(1:9)

# remove HP outliers
HEM_IBS_dat_adj <- HEM_IBS_dat %>% 
  filter(
    ! row_number() %in% HEM_IBS_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    )

# re-test for horizontal pleiotropy
mr_pleiotropy_test(HEM_IBS_dat_adj) %>% 
  select(-1:-2)

# perform MR
HEM_IBS_res <- mr(
  HEM_IBS_dat_adj
  )

# odds ratio
HEM_IBS_res <- generate_odds_ratios(
  HEM_IBS_res
  )

HEM_IBS_res %>%
  select(-1:-2)

# plot effects
mr_scatter_plot(
  HEM_IBS_res, HEM_IBS_dat_adj
)[[1]]

# leave one out analysis
HEM_IBS_loo <- mr_leaveoneout(
  HEM_IBS_dat_adj
  )

mr_leaveoneout_plot(HEM_IBS_loo)[[1]] +
  coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))

```

&nbsp;

### SNPs -> HEM (exposure) -> DIV (outcome)
```{r, message=TRUE, warning=FALSE}

# get effects of instruments on outcome
DIV_out_dat <- read_outcome_data(
    snps = HEM_exp_dat$SNP,
    filename = "data/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz",
    sep = " ",
    snp_col = "SNP",
    chr_col = "CHR", 
    pos_col = "BP",
    beta_col = "BETA",
    se_col = "SE",
    effect_allele_col = "ALLELE1",
    other_allele_col = "ALLELE0",
    eaf_col = "A1FREQ",
    pval_col = "P"
    ) %>% 
  # add outcome name
  mutate(
    outcome = "DIV (Schafmayer et al.)"
  )

# check if any SNPs are missing
missing <- DIV_out_dat %>% 
  filter(!SNP %in% HEM_exp_dat$SNP) %>% 
  pull(SNP)

missing

# harmonize the exposure and outcome data
HEM_DIV_dat <- harmonise_data(
  HEM_exp_dat,
  DIV_out_dat
  )

# Cochran's Q
mr_heterogeneity(HEM_DIV_dat) %>% 
  select(-1:-2)

# test for horizontal pleiotropy
mr_pleiotropy_test(HEM_DIV_dat) %>%
  select(-1:-2)

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
HEM_DIV_presso <- run_mr_presso(
  HEM_DIV_dat, 
  NbDistribution = 1000,
  SignifThreshold = 0.1
  )

# global test results
HEM_DIV_presso[[1]]$`MR-PRESSO results`$`Global Test`

# HP outliers
HEM_DIV_dat %>% 
  filter(
    row_number() %in% HEM_DIV_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    ) %>% 
  select(1:9)

# remove HP outliers
HEM_DIV_dat_adj <- HEM_DIV_dat %>% 
  filter(
    ! row_number() %in% HEM_DIV_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    )

# re-test for horizontal pleiotropy
mr_pleiotropy_test(HEM_DIV_dat_adj) %>% 
  select(-1:-2)

# perform MR
HEM_DIV_res <- mr(
  HEM_DIV_dat_adj
  )

# odds ratio
HEM_DIV_res <- generate_odds_ratios(
  HEM_DIV_res
  )

HEM_DIV_res %>%
  select(-1:-2)

# plot effects
mr_scatter_plot(
  HEM_DIV_res, HEM_DIV_dat_adj
)[[1]]

# leave one out analysis
HEM_DIV_loo <- mr_leaveoneout(
  HEM_DIV_dat_adj
  )

mr_leaveoneout_plot(HEM_DIV_loo)[[1]] +
  coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))

```

&nbsp;

## Reverse MR analysis

### SNPs -> IBS (exposure) -> HEM (outcome)
```{r, message=TRUE, warning=FALSE}

# get effects of instruments on outcome
IBS_exp_dat <- read_exposure_data(
  filename = "data/GCST90016564_buildGRCh37.tsv.gz",
  sep = "\t",
  snp_col = "variant_id",
  chr_col = "chromosome", 
  pos_col = "base_pair_location",
  beta_col = "beta",
  se_col = "standard_error",
  effect_allele_col = "effect_allele",
  other_allele_col = "other_allele",
  eaf_col = "effect_allele_frequency",
  pval_col = "p_value"
  ) %>% 
  filter(
    pval.exposure < 1e-05
  ) %>% 
  mutate(
    exposure = "IBS (Eijsbouts et al.)"
  )

# clump instruments
IBS_exp_dat <- clump_data(
  IBS_exp_dat,
  clump_r2 = 0.001,
  clump_kb = 1000
  )

# get effects of instruments on outcome
HEM_out_dat <- extract_outcome_data(
  snps = IBS_exp_dat$SNP,
  outcomes = "ebi-a-GCST90014033"
  ) %>% 
  mutate(
    outcome = "HEM (Zheng et al.)"
  )

# check if any SNPs are missing
missing <- HEM_out_dat %>% 
  filter(!SNP %in% IBS_exp_dat$SNP) %>% 
  pull(SNP)

missing

# harmonize the exposure and outcome data
IBS_HEM_dat <- harmonise_data(
  IBS_exp_dat,
  HEM_out_dat
  )

# Cochran's Q
mr_heterogeneity(IBS_HEM_dat) %>% 
  select(-1:-2)

# test for horizontal pleiotropy
mr_pleiotropy_test(IBS_HEM_dat) %>%
  select(-1:-2)

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
IBS_HEM_presso <- run_mr_presso(
  IBS_HEM_dat, 
  NbDistribution = 1000,
  SignifThreshold = 0.1
  )

# global test results
IBS_HEM_presso[[1]]$`MR-PRESSO results`$`Global Test`

# HP outliers
IBS_HEM_dat %>% 
  filter(
    row_number() %in% IBS_HEM_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    ) %>% 
  select(1:9)

# remove HP outliers
IBS_HEM_dat_adj <- IBS_HEM_dat %>% 
  filter(
    ! row_number() %in% IBS_HEM_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    )

# re-test for horizontal pleiotropy
mr_pleiotropy_test(IBS_HEM_dat_adj) %>% 
  select(-1:-2)

# perform MR
IBS_HEM_res <- mr(
  IBS_HEM_dat_adj
  )

# odds ratio
IBS_HEM_res <- generate_odds_ratios(
  IBS_HEM_res
  )

IBS_HEM_res %>%
  select(-1:-2)

# plot effects
mr_scatter_plot(
  IBS_HEM_res, IBS_HEM_dat_adj
)[[1]]

# leave one out analysis
IBS_HEM_loo <- mr_leaveoneout(
  IBS_HEM_dat_adj
  )

mr_leaveoneout_plot(IBS_HEM_loo)[[1]] +
  coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))

```
&nbsp;

### SNPs -> DIV (exposure) -> HEM (outcome)
```{r, message=TRUE, warning=FALSE}

# get effects of instruments on outcome
DIV_exp_dat <- read_exposure_data(
  filename = "data/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz",
  sep = " ",
  snp_col = "SNP",
  chr_col = "CHR", 
  pos_col = "BP",
  beta_col = "BETA",
  se_col = "SE",
  effect_allele_col = "ALLELE1",
  other_allele_col = "ALLELE0",
  eaf_col = "A1FREQ",
  pval_col = "P"
  ) %>% 
  filter(
    pval.exposure < 5e-08
  ) %>% 
  mutate(
    exposure = "DIV (Schafmayer et al.)"
  )

# clump instruments
DIV_exp_dat <- clump_data(
  DIV_exp_dat,
  clump_r2 = 0.001,
  clump_kb = 1000
  )

# get effects of instruments on outcome
HEM_out_dat <- extract_outcome_data(
  snps = DIV_exp_dat$SNP,
  outcomes = "ebi-a-GCST90014033"
  ) %>% 
  mutate(
    outcome = "HEM (Zheng et al.)"
  )

# check if any SNPs are missing
missing <- HEM_out_dat %>% 
  filter(!SNP %in% DIV_exp_dat$SNP) %>% 
  pull(SNP)

missing

# harmonize the exposure and outcome data
DIV_HEM_dat <- harmonise_data(
  DIV_exp_dat,
  HEM_out_dat
  )

# Cochran's Q
mr_heterogeneity(DIV_HEM_dat) %>% 
  select(-1:-2)

# test for horizontal pleiotropy
mr_pleiotropy_test(DIV_HEM_dat) %>%
  select(-1:-2)

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
DIV_HEM_presso <- run_mr_presso(
  DIV_HEM_dat, 
  NbDistribution = 1000,
  SignifThreshold = 0.1
  )

# global test results
DIV_HEM_presso[[1]]$`MR-PRESSO results`$`Global Test`

# HP outliers
DIV_HEM_dat %>% 
  filter(
    row_number() %in% DIV_HEM_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    ) %>% 
  select(1:9)

# remove HP outliers
DIV_HEM_dat_adj <- DIV_HEM_dat %>% 
  filter(
    ! row_number() %in% DIV_HEM_presso[[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
    )

# re-test for horizontal pleiotropy
mr_pleiotropy_test(DIV_HEM_dat_adj) %>% 
  select(-1:-2)

# perform MR
DIV_HEM_res <- mr(
  DIV_HEM_dat_adj
  )

# odds ratio
DIV_HEM_res <- generate_odds_ratios(
  DIV_HEM_res
  )

IBS_HEM_res %>%
  select(-1:-2)

# plot effects
mr_scatter_plot(
  DIV_HEM_res, DIV_HEM_dat_adj
)[[1]]

# leave one out analysis
DIV_HEM_loo <- mr_leaveoneout(
  DIV_HEM_dat_adj
  )

mr_leaveoneout_plot(DIV_HEM_loo)[[1]] +
  coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))

```

&nbsp;

### Conclusion

1) Inverse-variance weighted (IVW) of SNP>DIV>HEM shows that DIV significantly increase the incidence of HEM.

2) IVW of SNP>IBS>HEM shows that IBS significantly increase the incidence of HEM.

&nbsp;

