---
title: "Methodological refinement of Zhu *et al.* study"
output: rmarkdown::html_document
knit: (function(input, ...) {
  rmarkdown::render(
    input,
    output_dir = "../reports"
    )
  })
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

In the study by [Zhu *et al.*](https://gut.bmj.com/content/early/2023/01/23/gutjnl-2022-329307-0), the authors used correlated (in linkage disequilibrium) genetic variants as IVs, which violates assumptions of the IWV method [(Burgess and Thompson, 2017)](https://link.springer.com/article/10.1007/s10654-017-0255-x). Additionally, to remove horizontal pleiotropy, the authors used a threshold of P<0.05 to test for the association to the outcome on a genome-wide scale. Due to the high false positive rate of the P<0.05, this may lead to the removal of vertically pleiotropic variants.

Here, we refined MR analyses of Zhu *et al.* study, by employing a standard workflow for IV selection to ensure that the variants are uncorrelated as well as not horizontally pleiotropic.

&nbsp;

## Methods

### Instrument variable (IV) selection 

Refined thresholds used:
  
  1) Association with exposure (HEM GWAS) - P < 5e^-8^;
  2) Clumping for linkage disequilibrium (LD) - r^2^ < 0.001 within 1 Mb distance (1000G EUR);
  3) MR-PRESSO outlier test (P<0.05)

&nbsp;

### Mendelian Randomisation methods 

MR methods used:

  1) Inverse variance weighted (IVW);
  2) MR Egger;
  3) Weighted median; 
  4) Weighted mode;
  5) Simple mode

&nbsp;

### GWAS summary statistics data

Used in the Zhu *et al.* analysis:

* HEM ([Zheng *et al.*](https://pubmed.ncbi.nlm.nih.gov/33888516/)) - GCST90014033
* DIV ([Schafmayer *et al.*](https://pubmed.ncbi.nlm.nih.gov/30661054/)) - GCST008105
* DIV ([Dönertaş *et al.*](https://pubmed.ncbi.nlm.nih.gov/33959723/)) - GCST90038682
* IBS ([Dönertaş *et al.*](https://pubmed.ncbi.nlm.nih.gov/33959723/)) - GCST90038626
* IBS ([Jang et *al.*](https://pubmed.ncbi.nlm.nih.gov/34737426/)) - GCST90044173

&nbsp;

### Packages

MR analyses were performed using R packages [`TwoSampleMR`](https://mrcieu.github.io/TwoSampleMR/) and [`tidyverse`](https://www.tidyverse.org).

```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(TwoSampleMR)
```
&nbsp;

## Analysis and Results

&nbsp;

### Data download
```{r, message=FALSE, warning=FALSE, eval=FALSE}

ebi_ftp <- "http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/"

# DIV - GCST008105
# https://pubmed.ncbi.nlm.nih.gov/30661054/
download.file( 
  url = paste0(ebi_ftp,"GCST008001-GCST009000/GCST008105/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz"),
  destfile = "../data/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz",
  method = "curl"
)

# DIV - GCST90038682
# https://pubmed.ncbi.nlm.nih.gov/33959723/
download.file( 
  url = paste0(ebi_ftp,"GCST90038001-GCST90039000/GCST90038682/GCST90038682_buildGRCh37.tsv"),
  destfile = "data/GCST90038682_buildGRCh37.tsv",
  method = "curl"
)
# compress
R.utils::gzip(
  "data/GCST90038682_buildGRCh37.tsv",
  overwrite=TRUE
)

# IBS - GCST90038626
# https://pubmed.ncbi.nlm.nih.gov/33959723/
download.file( 
  url = paste0(ebi_ftp,"GCST90038001-GCST90039000/GCST90038626/GCST90038626_buildGRCh37.tsv"),
  destfile = "data/GCST90038626_buildGRCh37.tsv",
  method = "curl"
)
# compress
R.utils::gzip(
  "data/GCST90038626_buildGRCh37.tsv",
  overwrite=TRUE
)

# IBS - GCST90044173
# https://pubmed.ncbi.nlm.nih.gov/34737426/
download.file( 
  url = paste0(ebi_ftp,"GCST90044001-GCST90045000/GCST90044173/GCST90044173_buildGRCh37.tsv.gz"),
  destfile = "data/GCST90044173_buildGRCh37.tsv.gz",
  method = "curl"
)

# HEM - GCST90014033
# https://pubmed.ncbi.nlm.nih.gov/33888516/
# can be accessed directly from TwoSampleMR

```
&nbsp;

### HEM (exposure) and DIV (outcome)

&nbsp;

##### Obtain IVs from GWAS of exposure (HEM)

Here we obtain SNPs from HEM GWAS using clumping procedure with refined thresholds:
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# from OpenGWAS db
# thresholds as defined by Zhu et al.
HEM_exp_dat <- extract_instruments(
  "ebi-a-GCST90014033",
  p1 = 5e-8,
  r2 = 0.001,
  kb = 1000
  ) %>% mutate(
    exposure = "HEM (Zheng et al.)"
  )

# number of SNPs after clumping
nrow(HEM_exp_dat)

```
&nbsp;

##### Obtain effects of IVs on outcome (DIV)

Here we import DIV GWAS datasets (Schafmayer *et al.* and Dönertaş *et al.*) and filter the effects of SNPs, which  will be used as IVs.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# Schafmayer et al. data
DIV_1_out_dat <- read_outcome_data(
    snps = HEM_exp_dat$SNP,
    filename = "../data/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz",
    sep = " ",
    snp_col = "SNP",
    chr_col = "CHR", 
    pos_col = "BP",
    beta_col = "BETA",
    se_col = "SE",
    effect_allele_col = "ALLELE1",
    other_allele_col = "ALLELE0",
    eaf_col = "A1FREQ",
    pval_col = "P"
    ) %>% 
  # add outcome name
  mutate(
    outcome = "DIV (Schafmayer et al.)"
  )

# Dönertaş et al. data
DIV_2_out_dat <- read_outcome_data(
    snps = HEM_exp_dat$SNP,
    filename = "../data/GCST90038682_buildGRCh37.tsv.gz",
    sep = "\t",
    snp_col = "variant_id",
    chr_col = "chromosome", 
    pos_col = "base_pair_location",
    beta_col = "beta",
    se_col = "standard_error",
    effect_allele_col = "effect_allele",
    other_allele_col = "other_allele",
    eaf_col = "effect_allele_frequency",
    pval_col = "p_value"
    ) %>% 
  # add outcome name
  mutate(
    outcome = "DIV (Dönertaş et al.)"
  )

DIV_out_dat_list <- list(
  "DIV (Schafmayer et al.)" = DIV_1_out_dat,
  "DIV (Dönertaş et al.)" = DIV_2_out_dat
)

# check if any SNPs are missing
missing <- DIV_out_dat_list %>% 
  lapply(., function(x){
    x %>% filter(!SNP %in% HEM_exp_dat$SNP) %>% 
  pull(SNP)
  })
missing

```
&nbsp;

#### Harmonize exposure and outcome data

Here we harmonize SNPs, so the effects are relative to the same allele.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# harmonize data
HEM_DIV_dat_list <- lapply(DIV_out_dat_list, function(x){
  harmonise_data(HEM_exp_dat, x)
  })

# number of SNPs retained for MR analysis
lapply(HEM_DIV_dat_list, function(x){
  x %>% filter(mr_keep == TRUE) %>% 
    nrow()
})

```
&nbsp;

#### Test for horizontal pleiotropy

Here we use MR Egger intercept test to evaluate if selected IVs are not pleiotropic, thus have no effect on disease outside of its effect on the exposure.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# test for horizontal pleiotropy
lapply(HEM_DIV_dat_list, mr_pleiotropy_test) %>% 
  bind_rows() %>% select(-1:-2)

```
The results show that selected variants are no likely to be horizontally pleiotropic (P < 0.05).

&nbsp;

Here we use MR-PRESSO to detect and remove ouliers and then again run MR Egger intercept test to evaluate horizontal pleiotropy.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
HEM_DIV_presso_list <- lapply(
  HEM_DIV_dat_list,
  run_mr_presso,
  NbDistribution = 2000,
  SignifThreshold = 0.1
  )

# global test results
lapply(HEM_DIV_presso_list, function(x){
  x[[1]]$`MR-PRESSO results`$`Global Test`
})

# HP outliers
lapply(names(HEM_DIV_dat_list), function(i){
  HEM_DIV_dat_list[[i]] %>% 
    filter(
      row_number() %in% HEM_DIV_presso_list[[i]][[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
      ) %>% 
  select(1:9)
})

# remove HP outliers
HEM_DIV_dat_list_adj <- lapply(names(HEM_DIV_dat_list), function(i){
    HEM_DIV_dat_list[[i]] %>% 
    filter(
      ! row_number() %in% HEM_DIV_presso_list[[i]][[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
      ) 
})
names(HEM_DIV_dat_list_adj) <- names(HEM_DIV_dat_list)

# re-test for horizontal pleiotropy
lapply(HEM_DIV_dat_list_adj, mr_pleiotropy_test) %>% 
  bind_rows() %>% select(-1:-2)

```
The results show that selected variants are not likely to be horizontally pleiotropic (P > 0.05).

&nbsp;

#### Perform MR analyses 

Here, using the retained IVs, we perform MR by utilizing IVW, MR Egger, weighted median, simple mode and Weighted mode methods (default of the `TwoSampleMR` `mr()` function).
In the analysis by Zhu et al, the authors call MR analysis using DIV Schafmayer *et al.* and Dönertaş *et al.* datasets as "Set of experiment" and "Set of verification", respectively.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(5, 5)}

# perform MR
HEM_DIV_list_res <- lapply(
  HEM_DIV_dat_list_adj, mr
  )

# calculate odds ratio
HEM_DIV_list_res <- lapply(HEM_DIV_list_res, function(x){
  generate_odds_ratios(x)
})

# results
HEM_DIV_list_res %>% bind_rows() %>% 
  select(-1:-2)

# plot results
HEM_DIV_list_plot <- lapply(seq(HEM_DIV_list_res), function(i){
  mr_scatter_plot(HEM_DIV_list_res[[i]], HEM_DIV_dat_list_adj[[i]])
})
names(HEM_DIV_list_plot) <- names(HEM_DIV_list_res)

lapply(HEM_DIV_list_plot, function(x){
  x[[1]]
})
```

No statistically significant associations were identified in both analyses.

&nbsp;

#### Perform sensitivity analyses 

Here, we perform leave-one-out analysis to test if association was not driven by a single variant.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(8, 5)}
# leave one out analysis
HEM_DIV_list_loo <- lapply(
  HEM_DIV_dat_list_adj, mr_leaveoneout
  )

# plot results
lapply(HEM_DIV_list_loo, function(x){
  p <- mr_leaveoneout_plot(x)
  p[[1]] + coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))
  })

```
&nbsp;

### HEM (exposure) and IBS (outcome)

&nbsp;

##### Obtain effects of IVs on outcome (IBS)

Here we import IBS GWAS datasets (Dönertaş *et al.* and Jiang *et al.*) and filter the effects of SNPs, which  will be used as IVs.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# list files to IBS studies
IBS_file_list <- list(
  "IBS (Dönertaş et al.)" = "../data/GCST90038626_buildGRCh37.tsv.gz",
  "IBS (Jiang et al.)" = "../data/GCST90044173_buildGRCh37.tsv.gz"
)

# get effects of instruments on outcome
IBS_out_dat_list <- lapply(IBS_file_list, function(x){
  read_outcome_data(
    snps = HEM_exp_dat$SNP,
    filename = x,
    sep = "\t",
    snp_col = "variant_id",
    chr_col = "chromosome", 
    pos_col = "base_pair_location",
    beta_col = "beta",
    se_col = "standard_error",
    effect_allele_col = "effect_allele",
    other_allele_col = "other_allele",
    eaf_col = "effect_allele_frequency",
    pval_col = "p_value"
    )
  })

# add outcome name
for(i in seq(IBS_out_dat_list)){
  IBS_out_dat_list[[i]] <- IBS_out_dat_list[[i]] %>% 
    mutate(
      outcome = names(IBS_out_dat_list[i])
    )
}

# check if any SNPs are missing
missing <- IBS_out_dat_list %>% 
  lapply(., function(x){
    x %>% filter(!SNP %in% HEM_exp_dat$SNP) %>% 
  pull(SNP)
  })
missing

```
&nbsp;

#### Harmonize exposure and outcome data

As previously, we harmonize SNPs in exposure and outcome datasets.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# harmonize data
HEM_IBS_dat_list <- lapply(IBS_out_dat_list, function(x){
  harmonise_data(HEM_exp_dat, x)
  })

# number of SNPs retained for MR analysis
lapply(HEM_IBS_dat_list, function(x){
  x %>% filter(mr_keep == TRUE) %>% 
    nrow()
})

```
&nbsp;

#### Test for horizontal pleiotropy

As before, we use MR Egger intercept test to evaluate if selected IVs are not horizontally pleiotropic.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# test for horizontal pleiotropy
lapply(HEM_IBS_dat_list, mr_pleiotropy_test) %>% 
  bind_rows() %>% select(-1:-2)

```
&nbsp;

Here we use MR-PRESSO to detect and remove ouliers and then again run MR Egger intercept test to evaluate horizontal pleiotropy.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
HEM_IBS_presso_list <- lapply(
  HEM_IBS_dat_list,
  run_mr_presso,
  NbDistribution = 2000,
  SignifThreshold = 0.1
  )

# global test results
lapply(HEM_IBS_presso_list, function(x){
  x[[1]]$`MR-PRESSO results`$`Global Test`
})

```
Both methods show that the selected variants are not likely to be horizontally pleiotropic (P < 0.05).

&nbsp;

#### Perform MR analyses 

As previously, using the retained IVs, we perform MR by utilizing default methods of the `TwoSampleMR::mr()` function. As in case for DIV, the authors of the Zhu *et al.* call MR analysis using IBS Dönertaş *et al.* and Jiang *et al.* datasets as "Set of experiment" and "Set of verification", respectively.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(5, 5)}

# perform MR
HEM_IBS_list_res <- lapply(
  HEM_IBS_dat_list, mr
  )

# calculate odds ratio
HEM_IBS_list_res <- lapply(HEM_IBS_list_res, function(x){
  generate_odds_ratios(x)
})

# results
HEM_IBS_list_res %>% bind_rows() %>% 
  select(-1:-2)

# plot results
HEM_IBS_list_plot <- lapply(seq(HEM_IBS_list_res), function(i){
  mr_scatter_plot(HEM_IBS_list_res[[i]], HEM_IBS_dat_list[[i]])
})
names(HEM_IBS_list_plot) <- names(HEM_IBS_list_res)

lapply(HEM_IBS_list_plot, function(x){
  x[[1]]
})
```

As in case to DIV, no statistically significant associations were identified in both datasets.

&nbsp;

#### Perform sensitivity analyses 

Again, we perform leave-one-out analysis to test if association was not driven by a single variant.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(8, 5)}
# leave one out analysis
HEM_IBS_list_loo <- lapply(
  HEM_IBS_dat_list, mr_leaveoneout
  )

# plot results
lapply(HEM_IBS_list_loo, function(x){
  p <- mr_leaveoneout_plot(x)
  p[[1]] + coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))
  })

```
&nbsp;

### DIV (exposure) and HEM (outcome)

&nbsp;

##### Obtain effects of IVs on outcome (HEM)

Here we import DIV GWAS datasets (Schafmayer *et al.* and Dönertaş *et al.*) and use as exposure data.
```{r, message=FALSE, warning=FALSE}

# get effects of instruments on outcome
DIV_1_exp_dat <- read_exposure_data(
  filename = "../data/GWAS_summary_1-23.dosages.maf_0.01.info_0.4.txt.gz",
  sep = " ",
  snp_col = "SNP",
  chr_col = "CHR", 
  pos_col = "BP",
  beta_col = "BETA",
  se_col = "SE",
  effect_allele_col = "ALLELE1",
  other_allele_col = "ALLELE0",
  eaf_col = "A1FREQ",
  pval_col = "P"
  ) %>% 
  filter(
    pval.exposure < 5e-08
  ) %>% 
  mutate(
    exposure = "DIV (Schafmayer et al.)"
  )

DIV_2_exp_dat <- read_exposure_data(
  filename = "../data/GCST90038682_buildGRCh37.tsv.gz",
  sep = "\t",
  snp_col = "variant_id",
  chr_col = "chromosome", 
  pos_col = "base_pair_location",
  beta_col = "beta",
  se_col = "standard_error",
  effect_allele_col = "effect_allele",
  other_allele_col = "other_allele",
  eaf_col = "effect_allele_frequency",
  pval_col = "p_value"
  ) %>% 
  filter(
    pval.exposure < 1e-05
  ) %>% 
  mutate(
    exposure = "DIV (Dönertaş et al.)"
  )

DIV_exp_dat_list <- list(
  "DIV (Schafmayer et al.)" = DIV_1_exp_dat,
  "DIV (Dönertaş et al.)" = DIV_2_exp_dat
)

# add exposure name
for(i in seq(DIV_exp_dat_list)){
  DIV_exp_dat_list[[i]] <- DIV_exp_dat_list[[i]] %>% 
    mutate(
      exposure = names(DIV_exp_dat_list[i])
    )
}

# clump instruments
DIV_exp_dat_list <- lapply(
  DIV_exp_dat_list,
  clump_data,
  clump_r2 = 0.001,
  clump_kb = 1000
  )

# number of SNPs
lapply(DIV_exp_dat_list, nrow)

# get effects of instruments on outcome
HEM_out_dat_list <- lapply(DIV_exp_dat_list, function(x){
  extract_outcome_data(
    snps = x$SNP,
    outcomes = "ebi-a-GCST90014033"
    ) %>% 
    mutate(
      outcome = "HEM (Zheng et al.)"
      )
  })

# check if any SNPs are missing
missing <- lapply(seq(HEM_out_dat_list), function(i){
    DIV_exp_dat_list[[i]] %>% filter(!SNP %in% HEM_out_dat_list[[i]]$SNP) %>% 
    pull(SNP)
  })
names(missing) <- names(HEM_out_dat_list)
missing

```
&nbsp;

#### Harmonize exposure and outcome data

As previously, we harmonize SNPs in exposure and outcome datasets.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# harmonize the exposure and outcome data
DIV_HEM_dat_list <- lapply(seq(DIV_exp_dat_list), function(i){
  harmonise_data(DIV_exp_dat_list[[i]], HEM_out_dat_list[[i]])
})
names(DIV_HEM_dat_list) <- names(DIV_out_dat_list)

# number of SNPs retained for MR analysis
lapply(DIV_HEM_dat_list, function(x){
  x %>% filter(mr_keep == TRUE) %>% 
    nrow()
})

```
&nbsp;

#### Test for horizontal pleiotropy

As before, we use MR Egger intercept test to evaluate if selected IVs are not horizontally pleiotropic.
```{r, message=FALSE, warning=FALSE, eval=TRUE}
# test for horizontal pleiotropy
lapply(DIV_HEM_dat_list, mr_pleiotropy_test) %>% 
  bind_rows() %>% select(-1:-2)

```
&nbsp;

Here we use MR-PRESSO to detect and remove outliers and then again run MR Egger intercept test to evaluate horizontal pleiotropy.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
DIV_HEM_presso_list <- lapply(
  DIV_HEM_dat_list,
  run_mr_presso,
  NbDistribution = 2000,
  SignifThreshold = 0.1
  )

# global test results
lapply(DIV_HEM_presso_list, function(x){
  x[[1]]$`MR-PRESSO results`$`Global Test`
})

# HP outliers
lapply(names(DIV_HEM_dat_list), function(i){
  DIV_HEM_dat_list[[i]] %>% 
    filter(
      row_number() %in% DIV_HEM_presso_list[[i]][[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
      ) %>% 
  select(1:9)
})

# remove HP outliers
DIV_HEM_dat_list_adj <- lapply(names(DIV_HEM_dat_list), function(i){
    DIV_HEM_dat_list[[i]] %>% 
    filter(
      ! row_number() %in% DIV_HEM_presso_list[[i]][[1]]$`MR-PRESSO results`$`Distortion Test`$`Outliers Indices`
      ) 
})
names(DIV_HEM_dat_list_adj) <- names(DIV_HEM_dat_list)

# re-test for horizontal pleiotropy
lapply(DIV_HEM_dat_list_adj, mr_pleiotropy_test) %>% 
  bind_rows() %>% select(-1:-2)

```
The results show that the selected variants are not likely to be horizontally pleiotropic (P < 0.05).

&nbsp;

#### Perform MR analyses 

As previously, using the retained IVs, we perform MR by utilizing default methods of the `TwoSampleMR::mr()` function.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(5, 5)}

# perform MR
DIV_HEM_list_res <- lapply(
  DIV_HEM_dat_list_adj, mr
  )

# odds ratio
DIV_HEM_list_res <- lapply(DIV_HEM_list_res, function(x){
  generate_odds_ratios(x)
})

# results
DIV_HEM_list_res %>% bind_rows() %>% 
  select(-1:-2)

# plot results
DIV_HEM_list_plot <- lapply(seq(DIV_HEM_list_res), function(i){
  mr_scatter_plot(DIV_HEM_list_res[[i]], DIV_HEM_dat_list_adj[[i]])
})
names(DIV_HEM_list_plot) <- names(DIV_HEM_list_res)

lapply(DIV_HEM_list_plot, function(x){
  x[[1]]
})

```

In both of the datasets, we observe statistically significant associations ($P_{IWV}<0.05$; $bxy>0$) of DIV (exposure) on HEM (outcome).

&nbsp;

#### Perform sensitivity analyses 

Here we perform leave-one-out analysis to test if association was not driven by a single variant.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(5, 5)}
# leave one out analysis
DIV_HEM_list_loo <- lapply(
  DIV_HEM_dat_list_adj, mr_leaveoneout
  )

# plot results
lapply(DIV_HEM_list_loo, function(x){
  p <- mr_leaveoneout_plot(x)
  p[[1]] + coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))
  })

```

Sensitivity analyses show that there is no single variant/s which are driving the association.

&nbsp;

### IBS (exposure) and HEM (outcome)

&nbsp;

##### Obtain IVs from GWAS of exposure (IBS)

Here we obtain SNPs from the two IBS GWAS summary statistics using clumping procedure with refined thresholds:
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# get effects of instruments on outcome
IBS_exp_dat_list <- lapply(IBS_file_list, function(x){
  read_exposure_data(
    filename = x,
    sep = "\t",
    snp_col = "variant_id",
    chr_col = "chromosome", 
    pos_col = "base_pair_location",
    beta_col = "beta",
    se_col = "standard_error",
    effect_allele_col = "effect_allele",
    other_allele_col = "other_allele",
    eaf_col = "effect_allele_frequency",
    pval_col = "p_value"
    ) %>%
    filter(
      pval.exposure < 1e-05
      )
  })

# add exposure name
for(i in seq(IBS_exp_dat_list)){
  IBS_exp_dat_list[[i]] <- IBS_exp_dat_list[[i]] %>% 
    mutate(
      exposure = names(IBS_exp_dat_list[i])
    )
}

# clump instruments
IBS_exp_dat_list <- lapply(
  IBS_exp_dat_list,
  clump_data,
  clump_r2 = 0.001,
  clump_kb = 1000
  )

# number of SNPs
lapply(IBS_exp_dat_list, nrow)

# get effects of instruments on outcome
HEM_out_dat_list <- lapply(IBS_exp_dat_list, function(x){
  extract_outcome_data(
    snps = x$SNP,
    outcomes = "ebi-a-GCST90014033"
    ) %>% 
    mutate(
      outcome = "HEM (Zheng et al.)"
      )
  })

# check if any SNPs are missing
missing <- lapply(seq(HEM_out_dat_list), function(i){
    IBS_exp_dat_list[[i]] %>% filter(!SNP %in% HEM_out_dat_list[[i]]$SNP) %>% 
    pull(SNP)
  })
names(missing) <- names(HEM_out_dat_list)
missing

```
&nbsp;

#### Harmonize exposure and outcome data

As previously, we harmonize SNPs in exposure and outcome datasets.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# harmonize the exposure and outcome data
IBS_HEM_dat_list <- lapply(seq(IBS_exp_dat_list), function(i){
  harmonise_data(IBS_exp_dat_list[[i]], HEM_out_dat_list[[i]])
})
names(IBS_HEM_dat_list) <- names(HEM_out_dat_list)

# number of SNPs retained for MR analysis
lapply(IBS_HEM_dat_list, function(x){
  x %>% filter(mr_keep == TRUE) %>% 
    nrow()
})

```
&nbsp;

#### Test for horizontal pleiotropy

As before, we use MR Egger intercept test to evaluate if selected IVs are not horizontally pleiotropic.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# test for horizontal pleiotropy
lapply(IBS_HEM_dat_list, mr_pleiotropy_test) %>% 
  bind_rows() %>% select(-1:-2)

```
&nbsp;

Then again, we use MR-PRESSO to detect and remove outliers and then again run MR Egger intercept test to evaluate horizontal pleiotropy.
```{r, message=FALSE, warning=FALSE, eval=TRUE}

# set random seed
set.seed(0)

# run mr-presso to perform correction of
# horizontal pleiotropy via outlier removal
IBS_HEM_presso_list <- lapply(
  IBS_HEM_dat_list,
  run_mr_presso,
  NbDistribution = 2000,
  SignifThreshold = 0.1
  )

# global test results
lapply(IBS_HEM_presso_list, function(x){
  x[[1]]$`MR-PRESSO results`$`Global Test`
})

```
Both methods show that the selected variants are not likely to be horizontally pleiotropic (P < 0.05).

&nbsp;

#### Perform MR analyses 

As previously, using the retained IVs, we perform MR by utilizing default methods of the `TwoSampleMR::mr()` function.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(5, 5)}

# perform MR
IBS_HEM_list_res <- lapply(
  IBS_HEM_dat_list, mr
  )

# add odds ratio
IBS_HEM_list_res <- lapply(IBS_HEM_list_res, function(x){
  generate_odds_ratios(x)
})

# results
IBS_HEM_list_res %>% bind_rows() %>% 
  select(-1:-2)

# plot results
IBS_HEM_list_plot <- lapply(seq(IBS_HEM_list_res), function(i){
  mr_scatter_plot(IBS_HEM_list_res[[i]], IBS_HEM_dat_list[[i]])
})
names(IBS_HEM_list_plot) <- names(IBS_HEM_list_res)

lapply(IBS_HEM_list_plot, function(x){
  x[[1]]
})

```

In both of the datasets, we do not observe any statistically significant associations ($P_{IWV}<0.05$) of IBS (exposure) on HEM (outcome). However, in both of the datasets $bxy_{IWV}$ estimate is consistently positive ($bxy>0$).

&nbsp;

#### Perform sensitivity analyses 

Here we perform leave-one-out analysis to test if association was not driven by a single variant.
```{r, message=FALSE, warning=FALSE, eval=TRUE, fig.dim = c(5, 5)}
# leave one out analysis
IBS_HEM_list_loo <- lapply(
  IBS_HEM_dat_list, mr_leaveoneout
  )

# plot results
lapply(IBS_HEM_list_loo, function(x){
  p <- mr_leaveoneout_plot(x)
  p[[1]] + coord_flip() +
  theme(
    axis.text.x = element_text(
      angle = 90, vjust=0.5, hjust = 1, size = 6
    ))
  })

```

&nbsp;

In conclusion, by refining the parameters of analysis by Zhu *et al.* study, we could not replicate their findings. Moreover, by performing reverse MR analysis we show significant causal associations of DIV on HEM development.

&nbsp;

```{r, message=FALSE, warning=FALSE, eval=TRUE}
sessionInfo()
```
